fortex/health/processors/temporal_mention_normalizing_processor.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Temporal Mention Normalizer
+"""
+from typing import Dict, Set
+import importlib
+import re
+
+import spacy
+
+# pylint: disable=unused-import
+from timexy import Timexy
+
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+
+from ftx.medical.clinical_ontology import NormalizedTemporalForm
+
+__all__ = [
+    "TemporalMentionNormalizingProcessor",
+]
+
+
+class TemporalMentionNormalizingProcessor(PackProcessor):
+    r"""
+    Implementation of this TemporalMentionNormalizingProcessor has
+    been based on Timexy rule based model (based on spacy), A
+    rendition of it that exists on github has been referred
+    to as well.
+
+    Referred repository link:
+    https://github.com/paulrinckens/timexy
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.extractor = None
+
+    def set_up(self):
+        if self.configs.require_gpu:
+            spacy.require_gpu(self.configs.gpu_id)
+        if self.configs.prefer_gpu:
+            spacy.prefer_gpu(self.configs.gpu_id)
+        self.extractor = spacy.load(self.configs.model_name)
+        self.extractor.add_pipe(self.configs.pipe_name, before="ner")
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        TemporalMentionNormalizingProcessor is done on the basis
+        of using spacy, timexy and the corresponding trained model
+        for en_web_sm
+        """
+        path_str, module_str = self.configs.entry_type.rsplit(".", 1)
+        mod = importlib.import_module(path_str)
+        entry = getattr(mod, module_str)
+        for entry_specified in input_pack.get(entry_type=entry):
+            doc = self.extractor(entry_specified.text)
+            normalized_text = []
+            for e in doc.ents:
```

Comment:
Minor: I would suggest naming the iterators in a more meaningful manner, e can be entities and m can be matches on L82.